test/metabase/util/delay_test.clj

```diff
@@ -0,0 +1,13 @@
+(ns ^:mb/once metabase.util.delay-test
+  (:require [clojure.test :refer :all]
+            [metabase.util.delay :as delay]))
+
+(set! *warn-on-reflection* true)
+
+(deftest delay-with-ttl-test
+  (let [d (delay/delay-with-ttl 100 #(Object.))
+        vals1 (doall (repeatedly 20 (fn [] @d)))
```

Comment:
Why 20 and not 2 (for example)? I wonder how long before some strange virtual machine pauses for 100 ms between two calls...